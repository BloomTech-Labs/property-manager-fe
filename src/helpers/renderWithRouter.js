/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { render } from '@testing-library/react';
import {
  createHistory,
  createMemorySource,
  LocationProvider
} from '@reach/router';
import '@testing-library/jest-dom/extend-expect';

export default function renderWithRouter(
  ui,
  { route = '/', history = createHistory(createMemorySource(route)) } = {}
) {
  return {
    ...render(<LocationProvider history={history}>{ui}</LocationProvider>),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  };
}
