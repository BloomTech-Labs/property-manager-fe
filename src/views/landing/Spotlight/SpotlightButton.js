import React from 'react';
import { navigate } from '@reach/router';

const SpotlightButton = () => {
  return (
    <div className="btnWrapper">
      <button
        className="spotBtn"
        type="submit"
        onClick={() => navigate('/signup')}
      >
        Sign up
      </button>
    </div>
  );
};

export default SpotlightButton;
