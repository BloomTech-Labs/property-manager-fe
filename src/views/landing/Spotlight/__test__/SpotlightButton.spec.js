import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom';
import SpotlightButton from '../SpotlightButton';
import renderWithRouter from '../../../../helpers/renderWithRouter';

afterEach(rtl.cleanup);

test('Looks for a sign up text on the Button', async () => {
  const wrapper = rtl.render(<SpotlightButton />);
  expect(await wrapper.queryAllByText(/sign up/i));
});

test('Looks for a sign up text on the Button', async () => {
  const {
    container,
    getByText,
    history: { navigate }
  } = renderWithRouter(<SpotlightButton />);

  const leftClick = { button: 0 };
  rtl.fireEvent.click(getByText(/sign up/i), leftClick);
  await navigate('/signup');
  expect(await container.innerHTML).toMatch(/sign up/i);
});

test('snapshot', () => {
  const { container } = rtl.render(<SpotlightButton />);
  expect(container.firstChild).toMatchSnapshot();
});
