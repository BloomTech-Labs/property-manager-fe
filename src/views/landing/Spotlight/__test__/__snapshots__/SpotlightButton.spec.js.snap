// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="btnWrapper"
>
  <button
    class="spotBtn"
    type="submit"
  >
    Sign up
  </button>
</div>
`;
